cas-server-core/src/main/java/org/jasig/cas/ticket/TicketGrantingTicketImpl.java

```diff
@@ -151,6 +151,8 @@ public synchronized ServiceTicket grantServiceTicket(final String id,
         final List<Authentication> authentications = getChainedAuthentications();
         service.setPrincipal(authentications.get(authentications.size()-1).getPrincipal());
 
+        // remove this service if it already exists to only keep the latest service ticket for a service
+        this.services.values().remove(service);
```

Comment:
Could you ever log into Service A, but have two active sessions for A and thus two STs?